### PR TITLE
Imagery endpoint check

### LINKED ISSUE #27: [Images] "samples" endpoint should never be used for imagery telemetry
Requests for historical images sometimes use the /samples/ endpoint, and at other times does not. The `/samples` endpoint does not work for images (presumably because url strings cannot be easily aggregated).

### PR DESCRIPTION
**Pull request by jvigliotta**
Added openmct as a parameter to the historical telemetry provider so we can use telemetry api to check for imagery hints (technically could just replicate if there are any issues with that). If it is imagery, pass in imagery flag and do not use samples endpoint. This is combined with @nikhilmandlik updates for minmax since they're modifying same code.

closes #27 

### Author Checklist
Check | Passed
-- | --
Changes address original issue? | Y
Unit tests included and/or updated with changes? | N/A
Command line build passes? | Yes
Changes have been smoke-tested? | Partially Need to test with images on server
Testing instructions included? | Yes

### CODE REVIEW

File: src/providers/historical-telemetry-provider.js [hunk 1]
```diff
@@ -27,61 +27,55 @@ import {
...
+        this.openmct = openmct;
     }
 
     supportsRequest(domainObject) {
         return domainObject.type.startsWith('yamcs.');
     }
 
     request(domainObject, options) {
+        let metadata = this.openmct.telemetry.getMetadata(domainObject);
+        let isImagery = metadata.valuesForHints(['range']).length !== 0;
```
Comment (akhenry) [inline src/providers/historical-telemetry-provider.js:42]: There is actually an `imagery` hint for image telemetry. The imagery view itself [actually uses it](https://github.com/nasa/openmct/blob/59946e89ef57ce32f3c2298f20a5cd6def3c2ec1/src/plugins/imagery/ImageryViewProvider.js#L7).

**Review by akhenry — CHANGES_REQUESTED** [review]

File: src/providers/historical-telemetry-provider.js (hunk 1 above)
Comment (jvigliotta) [reply src/providers/historical-telemetry-provider.js:42]: ah man, I totally had that but had switched it out for testing locally on yamcs (no imagery locally). I'll revert that.

File: src/providers/historical-telemetry-provider.js [hunk 2]
```diff
@@ -27,61 +27,55 @@ import {
 } from '../utils.js';
 
 export default class YamcsHistoricalTelemetryProvider {
-    constructor(url, instance) {
+    constructor(url, instance, openmct) {
```
Comment (nikhilmandlik) [inline src/providers/historical-telemetry-provider.js:30]: just being too picky, prefer, `openmct` as first param. 😅

**Review by nikhilmandlik — APPROVED** [review]

File: src/providers/historical-telemetry-provider.js (hunk 2 above)
Comment (jvigliotta) [reply src/providers/historical-telemetry-provider.js]: take a look at my commit message you picky person... 💯

File: src/providers/object-provider.js [hunk 3]
```diff
@@ -208,7 +208,7 @@ export default class YamcsObjectProvider {
         }
 
         if (isAggregate) {
-            obj.type = 'folder';
+            obj.type = ;
```
Comment (akhenry) [inline src/providers/object-provider.js:211]: Something doesn't seem right here...

**Review by akhenry — CHANGES_REQUESTED** [review]
Seems to be a code error in some of the non-editable folder stuff, otherwise looks good.

File: src/providers/object-provider.js (hunk 3 above)
Comment (jvigliotta) [reply src/providers/object-provider.js:211]: Agreed! That's an odd one. Not sure how I would have missed that.

**Comment by jvigliotta** [comment]
@akhenry @nikhilmandlik going to wait to update my stuff here until Nikhil has pushed his min-max code.

**Comment by akhenry** [comment]
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? N/A - no unit tests for YAMCS plugin yet.
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
5. Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue) Y

**Pull request merged**